components/admin_console/permission_schemes_settings/guest_permissions_tree/guest_permissions_tree.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import Permissions from 'mattermost-redux/constants/permissions';
+import {Role} from 'mattermost-redux/types/roles';
+import {ClientLicense} from 'mattermost-redux/types/config';
+
+import PermissionGroup from '../permission_group.jsx';
+import EditPostTimeLimitButton from '../edit_post_time_limit_button';
+import EditPostTimeLimitModal from '../edit_post_time_limit_modal';
+
+type Props = {
+    license: ClientLicense;
+    onToggle: (role: string, permissions: string[]) => void;
+    readOnly: boolean;
+    scope: string;
+    selectRow: (permission: string) => void;
+    parentRole?: Role;
+    selected?: string | null;
+    role?: Partial<Role>;
+}
+
+const GuestPermissionsTree = ({license, onToggle, readOnly, scope, selectRow, parentRole, selected, role = {permissions: []}}: Props) => {
+    const [editTimeLimitModalIsVisible, setEditTimeLimitModalIsVisible] = React.useState(false);
+
+    const openPostTimeLimitModal = () => setEditTimeLimitModalIsVisible(true);
```

Comment:
Could you wrap all 3 of these in `useCallback`? Without that, we'll be passing new functions into the child components which'll cause them to re-render every time this does.

Reply:
Good point, I will keep in my mind for future submissions